bundler/lib/bundler/checksum.rb

```diff
@@ -2,6 +2,71 @@
 
 module Bundler
   class Checksum
+    class << self
+      def from_file_source(file_source, digest_algorithms: %w[sha256])
+        raise ArgumentError, "not a valid file source: #{file_source}" unless file_source.respond_to?(:with_read_io)
+
+        digests = digest_algorithms.map do |digest_algorithm|
+          [digest_algorithm.to_s, Bundler::SharedHelpers.digest(digest_algorithm.upcase).new]
+        end.to_h
+
+        file_source.with_read_io do |io|
+          until io.eof?
+            block = io.read(16_384)
```

Comment:
Can we make 16_384 constant and add comment?

Reply:
This is just cargo-culted from somewhere else. Other than being the "ultimate goal" in the game 2048, I suspect this is intended to use a common default block size, 16KiB. I'll call it `DEFAULT_BLOCK_SIZE`.